### PR TITLE
Add support for tuples in `OnNewAccount` hook

### PR DESCRIPTION
**Pull request by shawntabrizi**
This adds support for multiple modules to be specified with the `OnNewAccount` hook in the Balances module like so:

```
	type OnNewAccount = ((Module1, Module2), Module3);
```

or

```
	type OnNewAccount = (Module1, Module2, Module3);
```

Updates `OnFreeBalanceZero` to use the same `for_each_tuple` macro.

Fixes https://github.com/paritytech/substrate/issues/2763

### CODE REVIEW

File: srml/system/src/lib.rs [hunk 1]
```diff
@@ -105,6 +105,16 @@ pub trait OnNewAccount<AccountId> {
 impl<AccountId> OnNewAccount<AccountId> for () {
 	fn on_new_account(_who: &AccountId) {}
 }
+impl<
```
Comment (bkchr) [inline srml/system/src/lib.rs]: Please do as its done here: https://github.com/paritytech/substrate/blob/master/srml/timestamp/src/lib.rs#L200

Comment (shawntabrizi) [reply srml/system/src/lib.rs]: Should I also update this then?
https://github.com/paritytech/substrate/blob/master/srml/support/src/traits.rs#L73

Comment (bkchr) [reply srml/system/src/lib.rs]: Yeah

File: srml/support/src/traits.rs [hunk 2]
```diff
@@ -36,20 +38,24 @@ pub trait OnFreeBalanceZero<AccountId> {
...
-		Y::on_free_balance_zero(who);
+macro_rules! impl_on_free_balance_zero {
+	() => (
+		impl<AccountId> OnFreeBalanceZero<AccountId> for () {
+			fn on_free_balance_zero(_: &AccountId) {}
+		}
+	);
+
+	( $($t:ident)* ) => {
+		impl<AccountId: Clone, $($t: OnFreeBalanceZero<AccountId>),*> OnFreeBalanceZero<AccountId> for ($($t,)*) {
```
Comment (bkchr) [inline srml/support/src/traits.rs]:
```suggestion
		impl<AccountId, $($t: OnFreeBalanceZero<AccountId>),*> OnFreeBalanceZero<AccountId> for ($($t,)*) {
```

File: srml/system/src/lib.rs [hunk 3]
```diff
@@ -102,10 +102,24 @@ pub trait OnNewAccount<AccountId> {
...
-	fn on_new_account(_who: &AccountId) {}
+macro_rules! impl_on_new_account {
+	() => (
+		impl<AccountId> OnNewAccount<AccountId> for () {
+			fn on_new_account(_: &AccountId) {}
+		}
+	);
+
+	( $($t:ident)* ) => {
+		impl<AccountId: Clone, $($t: OnNewAccount<AccountId>),*> OnNewAccount<AccountId> for ($($t,)*) {
```
Comment (bkchr) [inline srml/system/src/lib.rs]:
```suggestion
		impl<AccountId, $($t: OnNewAccount<AccountId>),*> OnNewAccount<AccountId> for ($($t,)*) {
```

**Pull request merged**